bson/codec_options.py

```diff
@@ -327,6 +346,11 @@ def __new__(
         if not isinstance(type_registry, TypeRegistry):
             raise TypeError("type_registry must be an instance of TypeRegistry")
 
+        if not isinstance(datetime_conversion, str):
+            raise TypeError("datetime_conversion must be a string")
+        if datetime_conversion not in DatetimeConversionOpts.__members__.values():
+            raise ValueError(f"{datetime_conversion} is not a valid option for datetime_conversion")
```

Comment:
We can defer to the enum class for all the validation. For example:
```
datetime_conversion = DatetimeConversionOpts(datetime_conversion)
```

On a valid input:
```
>>> DatetimeConversionOpts("datetime")
<DatetimeConversionOpts.DATETIME: 'datetime'>
>>> DatetimeConversionOpts(_)
<DatetimeConversionOpts.DATETIME: 'datetime'>
```
On invalid input:
```
DatetimeConversionOpts("asd")
ValueError: 'asd' is not a valid DatetimeConversionOpts

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/enum.py", line 315, in __call__
    return cls.__new__(cls, value)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/enum.py", line 569, in __new__
    raise exc
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/enum.py", line 553, in __new__
    result = cls._missing_(value)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/enum.py", line 582, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 'asd' is not a valid DatetimeConversionOpts
```
On a non-string:
```
>>> DatetimeConversionOpts(2)
ValueError: 2 is not a valid DatetimeConversionOpts

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/enum.py", line 315, in __call__
    return cls.__new__(cls, value)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/enum.py", line 569, in __new__
    raise exc
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/enum.py", line 553, in __new__
    result = cls._missing_(value)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/enum.py", line 582, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 2 is not a valid DatetimeConversionOpts
```